test/orange/detail/BIHTraverser.test.cc

```diff
@@ -0,0 +1,202 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/BIHTraverser.test.cc
+//---------------------------------------------------------------------------//
+#include "orange/detail/BIHTraverser.hh"
+
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/data/CollectionMirror.hh"
+#include "orange/detail/BIHBuilder.hh"
+#include "orange/detail/BIHData.hh"
+#include "celeritas/Types.hh"
+
+#include "celeritas_test.hh"
+
+using BIHBuilder = celeritas::detail::BIHBuilder;
+using BIHInnerNode = celeritas::detail::BIHInnerNode;
+using BIHLeafNode = celeritas::detail::BIHLeafNode;
+using BIHTraverser = celeritas::detail::BIHTraverser;
+
+namespace celeritas
+{
+namespace test
+{
+class BIHTraverserTest : public Test
+{
+  public:
+    void SetUp() {}
+
+  protected:
+    std::vector<FastBBox> bboxes_;
+
+    BIHTreeData<Ownership::value, MemSpace::host> storage_;
+    BIHTreeData<Ownership::const_reference, MemSpace::host> ref_storage_;
+
+    static constexpr auto valid_volid_ = [](LocalVolumeId vol_id) -> bool {
+        return static_cast<bool>(vol_id);
+    };
+
+    static constexpr auto odd_volid_ = [](LocalVolumeId vol_id) -> bool {
+        return vol_id.unchecked_get() % 2 != 0;
+    };
```

Comment:
There's no need for these to be lambdas: just write them as static functions (free functions would work too)
```suggestion
    static constexpr bool valid_volid_(LocalVolumeId vol_id) {
        return static_cast<bool>(vol_id);
    };

    static constexpr bool odd_volid_(LocalVolumeId vol_id) {
        return vol_id.unchecked_get() % 2 != 0;
    };
```